src/main/java/com/fasterxml/jackson/core/util/BufferRecycler.java

```diff
@@ -189,4 +191,16 @@ protected int charBufferLength(int ix) {
 
     protected byte[] balloc(int size) { return new byte[size]; }
     protected char[] calloc(int size) { return new char[size]; }
+
+    BufferRecycler withPool(ObjectPool<BufferRecycler> pool) {
+        this._pool = pool;
```

Comment:
do we want to throw an exception in case is set twice? or will replace anything within?
or adding an assert to make evident the impl invariant here?